codalab/rest/bundles.py

```diff
@@ -736,7 +746,28 @@ def _update_bundle_contents_blob(uuid):
         if request.query.filename:
             filename = request.query.get('filename', default='contents')
             sources = [(filename, request['wsgi.input'])]
-        if sources:
+
+        if use_azure_blob_beta:
+            local.model.update_bundle(
+                bundle,
+                {
+                    'metadata': {
+                        # TODO(Ashwin): don't hardcode storage account name.
+                        'link_url': f"azfs://storageclwsdev0/bundles/{bundle.uuid}{'/contents.zip' if filename.endswith('.zip') else '/contents'}",
```

Comment:
I thought we were just going to zip up everything for consistency and to simplify the logic?

Reply:
No, we need to use `/contents` for single files, because otherwise we may not be able to distinguish between a bundle with a directory with a single file vs. a bundle with just a single file.